packages/react-server/src/ReactFlightServer.js

```diff
@@ -605,6 +622,33 @@ function renderClientElement(
   return element;
 }
 
+// The chunk ID we're currently rendering that we can assign debug data to.
+let debugID: null | number = null;
```

Comment:
This is mainly to know if we're at the shallow end of the task or deep within the task but we can use this for other things too.